### DISCUSSION TITLE
How about skipping coverage when tests fail and --bail is set to 1?

### DISCUSSION THREAD

**Discussion by capaj** [vitest-dev/vitest#3387] [Ideas]
as title says. I also created a feature proposal https://github.com/vitest-dev/vitest/issues/3388

**Comment by AriPerkkio** [comment]
I think we could easily implement this but as every other tool out there seems to report coverage even if tests fail we have been following same pattern: https://github.com/vitest-dev/vitest/issues/2908#issuecomment-1442890311. 

Personally I would be happy to see Vitest not reporting coverage in these cases. I don't think the `--bail` option should matter here.
